src/ucc/api/ucc.h

```diff
@@ -1981,6 +1981,9 @@ static inline ucc_status_t ucc_collective_test(ucc_coll_req_h request)
  *
  *  @ref ucc_collective_finalize operation releases all resources
  *  associated with the collective operation represented by the request handle.
+ *  In UCC_THREAD_MULTIPLE mode, the user is responsible for ensuring that
+ *  @ref ucc_collective_finalize is called after the status is UCC_OK and after
+ *  completing the execution of any callback registered with @ref ucc_coll_args_t.
  *
```

Comment:
how about returning status from callback to know if finalize() is called request is released so that we dont set externel status inside UCC.  so this way we can set external status after callback and i think we can remove this restriction